### PR TITLE
rust: Only install stable toolchain

### PR DESCRIPTION
**Pull request by chrysn**
Closes: https://github.com/RIOT-OS/riotdocker/issues/208

As of Rust 1.65, there is no need for nightly versions of Rust for RIOT any more.

Currently, this will likely (i.e. unless the nightly selection script happens to manage the case of no nightlies by assigning stable instead) not build, and depends on https://github.com/RIOT-OS/RIOT/pull/18839 which switches away from nightlies. (See there for an overview of all lockstepping involved)

### CODE REVIEW

**Comment by chrysn** [comment]
AIU backports don't use tagged CI builds, so this is stalled until the 2023.01 release makes 2022.10 unsupported.

(Unless we prefer to backport https://github.com/RIOT-OS/RIOT/pull/18839 -- that'd clear CI; release users, unlike CI, do use tagged dockers as far as I know. But my preference is to just let it grow out in 2023.01.)

**Comment by chrysn** [comment]
With https://github.com/RIOT-OS/RIOT/pull/18839 and https://github.com/RIOT-OS/RIOT/pull/19058 merged, we can now do this.

To make sure this is tested well, I'd prefer to merge this after https://github.com/RIOT-OS/riotdocker/pull/182. Either way, even when ACKed, I'd leave it to the current RM to merge.

**Comment by chrysn** [comment]
Now that all relevant changes are long in, *and* CI likes us again, I'd like to ask for final review on this.

(Also, the nightly version we had pinned is so ancient that it doesn't support features that riot-wrappers is starting to take for granted because they're stable already).

**Review by maribu — APPROVED** [review]

**Comment by chrysn** [comment]
Seems we're still not really in hard freeze yet, so squishing it in:

bors merge

**Pull request merged**
